tests/SpecTests/CommandExpectations.php

```diff
@@ -64,6 +64,18 @@ public static function fromCommandMonitoring(array $expectedEvents)
         return new self($expectedEvents);
     }
 
+    public static function fromCrud(array $expectedEvents)
+    {
+        $o = new self($expectedEvents);
+
+        /* The bulkWrite-arrayFilters test only includes the commandStarted
+         * event without asserting against the result of the command, so ignore
+         * any other events. */
+        $o->ignoreExtraEvents = true;
```

Comment:
I think you want:

```
$o->ignoreCommandFailed = true;
$o->ignoreCommandSucceeded = true;
```

Instead of this. Basically, `fromCommandMonitoring()` is currently the only spec that has assertions for commands failing/succeeding. Every other spec only asserts the command started event.

 `ignoreExtraEvents` is currently only used in the change stream spec, because its runner purposefully tells us to only check as many events are found in the `expectations` array. This is mainly so that it doesn't need to deal with variation around `getMore` commands across drivers; however, it's also a problem I'm trying to address with [SPEC-1341](https://jira.mongodb.org/browse/SPEC-1341) since we _would_ like to assert `getMore` in _some_ tests.

But I digress. For the CRUD tests, I don't think any `expectations` arrays are incomplete, so we should be able to leave `$ignoreExtraEvents` alone here so we can catch failures down the line (if we actually do end up missing an event).

Reply:
I've replaced `ignoreExtraEvents` with `ignoreCommandFailed` and `ignoreCommandSucceeded`. `ignoreExtraEvents` is a bit blunt since it slices off the list of commands that were stored. I think it's better to not store commands we don't care about in the first place.

Reply:
Looking at CommandExpectation with fresh eyes, the `array_slice()` is definitely odd. We do know the size of `expected` at the time the class is constructed, so it seems feasible that we could decide to ignore an event rather than continue appending to `$actualEvents`. Feel free to just refactor that in a later commit if you like (I don't think it warrants a review).